app/Services/Model/Imp/SummitService.php

```diff
@@ -3550,5 +3560,46 @@ public function generateQREncKey(Summit $summit):string
 
             return $enc_key;
         });
+
+        if (!is_null($res)) {
+            EncryptAllSummitBadgeQRCodes::dispatch($summit->getId());
+        }
+
+        return $res;
+    }
+
+    /**
+     * @param int $summit_id
+     * @return void
+     * @throws Exception
+     */
+    public function regenerateBadgeQRCodes(int $summit_id):void
+    {
+        $this->tx_service->transaction(function () use ($summit_id) {
```

Comment:
tx is to wide
please reduce the lock contention using a transaction per regneration ther repository should return the ids of the badges and on each tx u retrieve the badge per id and regenerate

Reply:
done